### PR TITLE
major refactor, clean up, and header bug fix

### PR DESCRIPTION
**Pull request by HusseinElguindi**
Hi, I removed unnecessary code and improved some logic, which in turn fixed some uncaught bugs. In the future, I would also recommend to deal with the data as bytes instead of converting it to strings. Hussein#9050.

### CODE REVIEW

File: proxy.go [hunk 1]
```diff
@@ -13,173 +13,148 @@ import (
...
-		case http.MethodPost:
-			fetchmode = "POST"
-		case http.MethodPut:
-			fetchmode = "PUT"
-		case http.MethodDelete:
-			fetchmode = "DELETE"
-		case http.MethodOptions:
-			fetchmode = "OPTIONS"
-		default:
-			fetchmode = "GET"
```
Comment (HusseinElguindi) [inline proxy.go:47]: r.Method is already implemented as a string, no need for the switch.

File: proxy.go [hunk 2]
```diff
@@ -13,173 +13,148 @@ import (
...
+	// Change JA3
+	// Use Chrome JA3 by default
+	tlsClient := "771,4865-4867-4866-49195-49199-52393-52392-49196-49200-49162-49161-49171-49172-51-57-47-53-10,0-23-65281-10-11-35-16-5-51-43-13-45-28-21,29-23-24-25-256-257,0"
+	if strings.Contains(strings.ToLower(userAgent), "firefox") {
+		tlsClient = "771,4865-4867-4866-49195-49199-52393-52392-49196-49200-49162-49161-49171-49172-156-157-47-53-10,0-23-65281-10-11-35-16-5-51-43-13-45-28-21,29-23-24-25-256-257,0"
 	}
-	cond1 := (len(r.Header["Poptls-Url"]) > 0)
-	cond2 := (len(r.Header["User-Agent"]) > 0)
-	proceed := cond1 && cond2
-	if proceed {
```
Comment (HusseinElguindi) [inline proxy.go:64]: unnecessary logic.

**Comment by Carcraftz** [comment]
LGTM! Thanks!

I'm currently working on a modified version of the cycletls library which should enable me to support more features such as not allowing redirects, header order, and passing body as bytes. Will probably implement that when I'm done

**Comment by HusseinElguindi** [comment]
> LGTM! Thanks!
> 
> 
> 
> I'm currently working on a modified version of the cycletls library which should enable me to support more features such as not allowing redirects, header order, and passing body as bytes. Will probably implement that when I'm done

That sounds good, I'd be happy to contribute!

**Pull request merged**